lib/services.js

```diff
@@ -191,6 +207,26 @@ export default {
                  });
     },
 
+   /**
+    * Check that hashedStream.completedHash matches header contentMd5.
+    * @param {object} request - request linked to the stream containine the data
+    * @param {string} hashedStream - hased stream once completed
+    * {key, dataStoreName (mem/file)}
+    * @param {RequestLogger} log - the current request logger
+    * @return {object} errors
+    */
+    checkHashMatchMD5(request, hashedStream, log) {
+        if (request.contentMD5 && hashedStream.completedHash &&
```

Comment:
why would we not have a request.contentMD5 or hashedStream.completedHash at this point?
1) I believe we are checking request[content-md5] and setting the value to request.contentMD5 in the route so we should have it?
2) if we don't have a hashedStream.completedHash at this point we should return an errors.InternalError.